nf-core/differentialabundance
# Healx starting differential abundance workflow

For demonstration and basis for discussion, this WIP PR demonstrates my take on a differential abundance workflow using various recently developed modules. 

One or two modules are still awaiting approval, others are awaiting PR approvals for fixes, but all should work fine as committed here. I run locally like:

```bash
nextflow run -resume -profile mamba main.nf \
    --input $(pwd)/testdata/SRP254919.samplesheet.csv \
    --gtf $(pwd)/testdata/genes.gtf.gz \
    --contrasts $(pwd)/testdata/SRP254919.contrasts.csv \
    --matrix $(pwd)/testdata/SRP254919.salmon.merged.gene_counts.top1000cov.tsv \
    --outdir $(pwd)/testdata/output
```

All test files are available in the test data repo, apart from the GTF, which I retrieved from iGenomes for mouse. 

The steps are:

- Make a feature annotation table from a GTF
- Make a feature/ observation / matrix composite from the features, samples and input matrix (I see this as being transferable to features other than genes in future).
- Run a validation to check the internal consistency of features, samples, matrix and contrasts
- Run differential expression analysis per contrast with DESeq2
- Run an exploratory analysis on matrix outputs, with separate coloring for each unique variable used to define contrasts (see notes in workflow comment)
- Generate volcano plots per contrast. 

The output file structure is like:

```
testdata/output
├── pipeline_info
│   ├── execution_report_2022-10-28_23-00-55.html
│   ├── execution_timeline_2022-10-28_23-00-55.html
│   ├── execution_trace_2022-10-28_23-00-55.txt
│   ├── pipeline_dag_2022-10-28_23-00-55.html
│   └── software_versions.yml
├── plots
│   ├── differential
│   │   ├── treatment_mCherry_hND6_
│   │   │   ├── html
│   │   │   └── png
│   │   ├── treatment_mCherry_hND6_sample_number
│   │   │   ├── html
│   │   │   └── png
│   │   └── versions.yml
│   └── exploratory
│       ├── treatment
│       │   ├── html
│       │   └── png
│       └── versions.yml
└── tables
    └── differential
        ├── treatment-mCherry-hND6-sample_number.R_sessionInfo.log
        ├── treatment-mCherry-hND6-sample_number.dds.rld.rds
        ├── treatment-mCherry-hND6-sample_number.deseq2.dispersion.png
        ├── treatment-mCherry-hND6-sample_number.deseq2.results.tsv
        ├── treatment-mCherry-hND6-sample_number.deseq2.sizefactors.tsv
        ├── treatment-mCherry-hND6-sample_number.normalised_counts.tsv
        ├── treatment-mCherry-hND6-sample_number.vst.tsv
        ├── treatment-mCherry-hND6.R_sessionInfo.log
        ├── treatment-mCherry-hND6.dds.rld.rds
        ├── treatment-mCherry-hND6.deseq2.dispersion.png
        ├── treatment-mCherry-hND6.deseq2.results.tsv
        ├── treatment-mCherry-hND6.deseq2.sizefactors.tsv
        ├── treatment-mCherry-hND6.normalised_counts.tsv
        ├── treatment-mCherry-hND6.vst.tsv
        └── versions.yml

15 directories, 22 files
```

## To do

- [ ] Gather feedback
- [x] Get the actual CI up and running using the available test data
- [ ] Start to account for non-RNA-seq data  
- [ ] Better integrated reporting (MultiQC integration?)

## PR checklist

- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
